File: audience/src/main/java/br/com/concretesolutions/audience/core/actor/Actor.java
package br.com.concretesolutions.audience.core.actor;

/**
 * Defines an instance to be an actor.
 */
public interface Actor {

    /**
     * Called when the instance is registered on the system.
     *
     * @param thisRef This actor's reference
     */
    void onActorRegistered(ActorRef thisRef);

}

File: audience-core/src/main/kotlin/io/github/victorolinasc/audience/core/actor/Actor.kt
package io.github.victorolinasc.audience.core.actor

/**
 * Defines an instance to be an actor.
 */
interface Actor {

    /**
     * Called when the instance is registered on the system.
     *
     * @param thisRef This actor's reference
     */
    fun onActorRegistered(thisRef: ActorRef)

}
